lib/std/wrapnils.nim

```diff
@@ -0,0 +1,42 @@
+type Wrapnil*[T] = object
+  valueImpl*: T
+
+proc wrapnil*[T](a: T): Wrapnil[T] =
+  ## Allows chains of field-access and indexing where the LHS can be nil.
+  ## This simplifies code by reducing need for if-else branches around intermediate
+  ## maybe nil values.
+  ## Note: experimental module and relies on {.experimental: "dotOperators".}
```

Comment:
I think this should be at the top of the module no?

Reply:
done